File: api/src/main/java/br/com/furb/domain/dto/UserInformationsDTO.java
package br.com.furb.domain.dto;

import br.com.furb.domain.Institution;
import br.com.furb.domain.Volunteer;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class UserInformationsDTO {

	private UserDTO user;
	private Volunteer volunteer;
	private Institution institution;
	private List<SocialProjectDTO> socialProjects;

}

File: app/src/app/domain/user-informations-dto.ts
import { User } from "../security/auth/user";
import { Volunteer } from "./volunteer";
import { Institution } from "./institution";
import { SocialProjectDTO } from "./social-project-dto";

export class UserInformationsDTO {
    user: User;
    volunteer: Volunteer;
    institution: Institution;
    socialProjects: SocialProjectDTO[];
}